### ISSUE TITLE
Formatter --check doesn't seem to work

### ISSUE THREAD

**Issue by ghiscoding** [oxc-project/oxc#17441]
Running Formatter --check option doesn't seem to work, it's twice now that new versions caused issues that weren't detected by the --check option. The last oxfmt version included changes that removed newline at the end of each of my files but it wasn't detected as shown below:

```
> oxfmt . --check

Checking formatting...

All matched files use the correct format.
Finished in 359ms on 366 files using 4 threads.
```

It always return no errors, but it happen twice now that new `oxfmt` version causes format changes and the `--check` didn't detect them. The output shown above is one of those time that it doesn't detect any problems, for reference here's the lint CI workflow, https://github.com/lerna-lite/lerna-lite/actions/runs/20541466864/job/59006353342, that should have detected the issue, but when I tried to publish a new version in this CI workflow (https://github.com/lerna-lite/lerna-lite/actions/runs/20541511031/job/59006452776) but failed because Lerna-Lite won't publish when changes are detected (which happen because the publish command always run an oxfmt format)... so, I had to fix the formatting by this commit (https://github.com/lerna-lite/lerna-lite/commit/4e6fc60ade2a91c4137739180f976922ef747494) which clearly shows some changes that the --check should have detected but it didn't...

So it would be great if the `oxfmt . --check` would correctly detect when format changes are required

**Comment by Boshen** [comment]
Investigating. Local works but not CI.

**Comment by Boshen** [comment]
Almost had a heart attack when I see critical errors like this haha

Check your log, you ran `pnpm format` in `build:full`

```
    "build:full": "pnpm clean && pnpm lint && pnpm format && tsgo --build ./tsconfig.packages.json && echo TSC Build Completed",
```

https://github.com/lerna-lite/lerna-lite/blob/fe6be748fa0f9ac581c446def4a2bdf8271fe6d9/package.json#L22

---

your pnpm cache is also not working properly, try the technique in https://github.com/oxc-project/setup-node

**Comment by ghiscoding** [comment]
> Almost had a heart attack when I see critical errors like this haha
> 
> Check your log, you ran `pnpm format` in `build:full`

oh damn that was silly of me, indeed the full build script did run the formatter prior to the --check, which is why it didn't detect anything... bad logic from my part 😓 

> your pnpm cache is also not working properly, try the technique in https://github.com/oxc-project/setup-node

Maybe I'm wrong but I don't think we should use any cache in a release workflow!? Also FYI (for your info),  with the latest version of setup-node, we can simply use `cache: 'pnpm`` instead of having to retrieve the store directory which is what I use in my main PR workflow but not in my release workflow (because again I'm not sure we should use cache in a release workflow?)

```
- name: Set NodeJS
    uses: actions/setup-node@v6
    with:
      node-version: 24
      cache: 'pnpm'
```

So with the new `cache` property, we don't need to read the store directory to create a cache key anymore, you can read their updated docs: 
https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data
